### ISSUE TITLE
Nuxt server components not working in SWR mode

### ISSUE THREAD

**Issue by mcrapts** [nuxt/nuxt#24088]
### Environment

``` 
------------------------------
- Operating System: Linux
- Node Version:     v18.18.0
- Nuxt Version:     3.8.0
- CLI Version:      3.9.1
- Nitro Version:    2.7.2
- Package Manager:  npm@9.4.2
- Builder:          -
- User Config:      devtools, routeRules, experimental
- Runtime Modules:  -
- Build Modules:    -
------------------------------
```

### Reproduction

https://stackblitz.com/edit/github-99tvkt?file=nuxt.config.ts

### Describe the bug

Server components do not render with `swr` enabled. See the attached reproduction.

**Comment by bernhardberger** [comment]
I'm having the same problem. Wanted to get rid of a lot the unnecessary hydration (most of my site doesn't need client side JS and just displays static data) and everything was looking good and pronising on devServer and with SSR build. But after turning on SWR all went to hell and most of the components just refused to render after the first SWR hit..

**Comment by bernhardberger** [comment]
Something iffy is going on with this reproduction as I the issue doesn't seem to be reproducible on stackblitz? Even when pinning the nuxt version to 3.8.0

**Comment by bernhardberger** [comment]
okay, pinning `nitropack` to 2.7.2 breaks it again.. so this seems to have been fixed in nitro at some point (not too far into the past as I had real world problems before launching a new site in late June).

I'll try to pinpoint the nitro version this seems to have been fixed.

**Comment by bernhardberger** [comment]
Okay, this issue seems to have vaporized itself (at least in the reproduction) with `nitro 2.9.0` release. No mentions in any changelog whatsoever though..

Broken (nuxt 3.8.0 & nitro <= 2.8.1):
https://stackblitz.com/edit/github-99tvkt-klrzvv?file=package.json

Working (nuxt 3.8.0 and nitro >= 2.9.0):
https://stackblitz.com/edit/github-99tvkt-s9jhf5?file=package.json

@danielroe I think this issue can be closed. But I also think this should be mentioned **somewhere** that this has implicitly been fixed with nitro 2.9.0 release as there's a bunch of folks (including me) that steered clear of using server components when using SWR because it was broken until recently and there was no mention anywhere that this has seemingly been fixed.
